### PR TITLE
Feat/wallet services

### PR DESCRIPTION
**Pull request by grvgoel81**
* Wallet services code added in general and for Animoca.
* Enable MFA.
* Webview for wallet services.

https://github.com/Web3Auth/web3auth-swift-sdk/assets/44719313/c72bb191-226b-47fb-ad17-e8e6b7492bc3

### CODE REVIEW

File: Sources/Web3Auth/Web3Auth.swift [hunk 1]
```diff
@@ -210,11 +380,11 @@ public class Web3Auth: NSObject {
...
     }
 
-    static func decodeStateFromCallbackURL(_ callbackURL: URL) throws -> Web3AuthState {
-        guard
+    static func decodeStateFromCallbackURL(_ callbackURL: URL) throws -> SessionResponse {
+       guard
             let callbackFragment = callbackURL.fragment,
-            let callbackData = Data.fromBase64URL(callbackFragment),
-            let callbackState = try? JSONDecoder().decode(Web3AuthState.self, from: callbackData)
+            let callbackData = Data.fromBase64URL(callbackFragment.components(separatedBy: "b64Params=")[1]),
```
Comment (chaitanyapotti) [inline Sources/Web3Auth/Web3Auth.swift:386]: fix parsing url

File: Sources/Web3Auth/WebViewController.swift [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+        preferences.javaScriptEnabled = true
+        preferences.javaScriptCanOpenWindowsAutomatically = true
+
+        let configuration = WKWebViewConfiguration()
+        configuration.preferences = preferences
+        configuration.applicationNameForUserAgent = "Version/8.0.2 Safari/600.2.5"
+
+        webView = WKWebView(frame: view.bounds, configuration: configuration)
+        webView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        webView.customUserAgent = "Mozilla/5.0 (Linux; Android 8.0; Pixel 2 Build/OPD3.170816.012) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/93.0.4577.82 Mobile Safari/537.36"
```
Comment (chaitanyapotti) [inline Sources/Web3Auth/WebViewController.swift:46]: no need

File: Sources/Web3Auth/WebViewController.swift [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+        view.addSubview(webView)
+    }
+    
+    func webView(_ webView: WKWebView, decidePolicyFor navigationAction: WKNavigationAction, decisionHandler: @escaping (WKNavigationActionPolicy) -> Void) {
+            if let redirectUrl = redirectUrl, !redirectUrl.isEmpty {
+                if let url = navigationAction.request.url, url.absoluteString.contains(redirectUrl) {
+                    let uri = URL(string: url.absoluteString)
+                    let host = uri?.host ?? ""
+                    let fragment = uri?.fragment ?? ""
+                    let component = fragment.components(separatedBy: "=")
```
Comment (chaitanyapotti) [inline Sources/Web3Auth/WebViewController.swift:59]: parse url correctly

File: Web3authSwiftSdkDemo/Web3authSwiftSdkDemo/ViewModel.swift [hunk 4]
```diff
@@ -21,18 +22,33 @@ class ViewModel: ObservableObject {
     var errorMessage: String = ""
     private var clientID: String = "BG4pe3aBso5SjVbpotFQGnXVHgxhgOxnqnNBKyjfEJ3izFvIVWUaMIzoCrAfYag8O6t6a6AOvdLcS4JR2sQMjR4"
     private var network: Network = .sapphire_devnet
-    private var buildEnv: BuildEnv = .staging
+    private var buildEnv: BuildEnv = .testing
   //  private var clientID: String = "BEaGnq-mY0ZOXk2UT1ivWUe0PZ_iJX4Vyb6MtpOp7RMBu_6ErTrATlfuK3IaFcvHJr27h6L1T4owkBH6srLphIw"
   //  private var network: Network = .mainnet
     private var useCoreKit: Bool = false
+    private var chainConfig: ChainConfig = ChainConfig(
+        chainNamespace: ChainNamespace.eip155,
```
Comment (chaitanyapotti) [inline Web3authSwiftSdkDemo/Web3authSwiftSdkDemo/ViewModel.swift:30]: make decimals and ticker nullable

File: Sources/Web3Auth/WebViewController.swift (hunk 3 above)
Comment (grvgoel81) [reply Sources/Web3Auth/WebViewController.swift]: resolved

**Review by chaitanyapotti — APPROVED** [review]
lgtm

**Pull request merged**